libs/react-client/src/store/utils.ts

```diff
@@ -0,0 +1,25 @@
+export interface SetterFn<T> {
+  (
+    partial: T | Partial<T> | ((state: T) => T | Partial<T>),
+    replace?: false
+  ): void;
+  (state: T | ((state: T) => T), replace: true): void;
+}
+
+export const stateOrSetter = <T, K extends keyof T>(
```

Comment:
Why do we need this helper, if `set` already support this?
<img width="996" height="100" alt="image" src="https://github.com/user-attachments/assets/ff47f349-4694-4053-82d1-f4840ae2a441" />